deepspeed/runtime/engine.py

```diff
@@ -114,14 +114,10 @@
 
 def split_half_float_double_sparse(tensors):
     device_type = get_accelerator().device_name()
-    supported_types = [
-        "torch.{}.HalfTensor".format(device_type), "torch.{}.FloatTensor".format(device_type),
-        "torch.{}.DoubleTensor".format(device_type), "torch.{}.BFloat16Tensor".format(device_type),
-        SparseTensor.type()
-    ]
-
+    supported_types = get_accelerator().supported_dtypes()
+    
     for t in tensors:
-        assert t.type() in supported_types, f"attempting to reduce an unsupported grad type: {t.type()}"
+        assert t.dtype in supported_types, f"attempting to reduce an unsupported grad type: {t.dtype}"
```

Comment:
I notice that `t.type()` is still used in line 124. This wouldn't work.

Reply:
Could you elaborate on this? t.type() was replaced with t.dtype
thanks

Reply:
Sure, I mean the below line 124
![image](https://github.com/microsoft/DeepSpeed/assets/4271600/389d2546-f307-4bf7-b09a-55bfaa6ba758)